### PR TITLE
Fix: Use SUPPORTS_INSERT_OVERWRITE and cleanup

### PR DESCRIPTION
**Pull request by eakmanrq**
Depends on: https://github.com/TobikoData/sqlmesh/pull/890

### CODE REVIEW

File: sqlmesh/core/engine_adapter/base.py [hunk 1]
```diff
@@ -596,13 +596,38 @@ def _insert_overwrite_by_condition(
...
+            query = t.cast("Query", query_or_df)
+            if where is not None:
+                query = (
+                    exp.select("*")
+                    .from_(
+                        t.cast(exp.Subqueryable, query).subquery("_subquery", copy=False),
+                        copy=False,
+                    )
+                    .where(where)
+                )
```
Comment (eakmanrq) [inline sqlmesh/core/engine_adapter/base.py:619]: @tobymao Is this the right way to add a where clause?

File: sqlmesh/core/engine_adapter/base.py [hunk 2]
```diff
@@ -596,13 +596,38 @@ def _insert_overwrite_by_condition(
...
+                    pandas_to_sql(
+                        df,
+                        alias=table.alias_or_name,
+                        columns_to_types=columns_to_types,
+                    )
+                )
+            query = t.cast("Query", query_or_df)
+            if where is not None:
+                query = (
+                    exp.select("*")
```
Comment (GeorgeSittas) [inline sqlmesh/core/engine_adapter/base.py:613]:
```suggestion
                    exp.select("*", copy=False)
```

File: sqlmesh/core/engine_adapter/base.py [hunk 3]
```diff
@@ -596,13 +596,38 @@ def _insert_overwrite_by_condition(
...
+                    .from_(
+                        t.cast(exp.Subqueryable, query).subquery("_subquery", copy=False),
```
Comment (GeorgeSittas) [inline sqlmesh/core/engine_adapter/base.py:615]: Let's be explicit about setting the `alias` kwarg here:
```suggestion
                        t.cast(exp.Subqueryable, query).subquery(alias="_subquery", copy=False),
```

File: sqlmesh/core/engine_adapter/base.py (hunk 1 above)
Comment (GeorgeSittas) [reply sqlmesh/core/engine_adapter/base.py:619]: Looks good: this will _append_ the `where` filter and copy it (by default).

**Comment by GeorgeSittas** [comment]
Seems like we're using SQLGlot builders without providing `dialect = self.dialect`. Is this safe?

File: sqlmesh/core/engine_adapter/base.py (hunk 1 above)
Comment (tobymao) [reply sqlmesh/core/engine_adapter/base.py:619]: the where is just used for deletes, so this is unnecessary, also building the query in this way won't work due to CTEs and other complexity

**Review by tobymao — APPROVED** [review]

**Pull request merged**